halo2_proofs/src/multicore.rs

```diff
@@ -3,3 +3,14 @@
 //! be extended in the future to allow for various parallelism strategies.
 
 pub use rayon::{current_num_threads, scope, Scope};
+
+pub(crate) fn log_threads() -> u32 {
+    let num_threads = current_num_threads();
+    assert!(num_threads > 0);
+    let mut pow = 0;
+    while (1 << (pow + 1)) <= num_threads {
+        pow += 1;
+    }
+
+    pow
```

Comment:
This can be written as
```
let num_threads = current_num_threads();
assert!(num_threads > 0 && num_threads <= u64::MAX);
63 - (num_threads as u64).leading_zeros()
```

Reply:
This is great.